tests/unit/CodeBox.spec.ts

```diff
@@ -0,0 +1,165 @@
+    expect(words.length).toBe(code.words.filter((word) => word.word != WordType.TAB && word.word != WordType.NEWLINE).length);
+
+    const spaces = wrapper.findAll('.code-space');
+    expect(spaces.length).toBe(code.words.filter((word) => word.word == WordType.SPACE).length);
+  });
+  test('CodeBox shows code correct after new props rendered', async () => {
+    const newCode = exampleCode2();
+    await wrapper.setProps({ code: newCode });
+
+    const lines = wrapper.findAll('.code-line');
+    expect(lines.length).toBe(newCode.words.filter((word) => word.word == WordType.NEWLINE).length + 1);
+
+    const words = wrapper.findAll('.code-word');
+    expect(words.length).toBe(newCode.words.filter((word) => word.word != WordType.TAB && word.word != WordType.NEWLINE).length);
+
+    const spaces = wrapper.findAll('.code-space');
+    expect(spaces.length).toBe(newCode.words.filter((word) => word.word == WordType.SPACE).length);
+  });
+  test('CodeBox shows no feedback when there is none feedback provided', () => {
+    const rightCodes = wrapper.findAll('.right-code');
+    expect(rightCodes.length).toBe(0);
+
+    const wrongCodes = wrapper.findAll('.wrong-code');
+    expect(wrongCodes.length).toBe(0);
+  });
+  test('CodeBox shows feedback when there is feedback provided', async () => {
+    const successWordFeedback = new WordFeedback(new Bug(2, ErrorType.LEXICAL, 'public'), true, true, true, true);
+    const failedWordBeedback = new WordFeedback(
+      new Bug(12, ErrorType.SYNTAX, 'System.out.println("HELLO THERE!");'),
+      false,
+      false,
+      false,
+      false
+    );
+    const newCodeFeedback = new CodeFeedback([successWordFeedback, failedWordBeedback]);
+    await wrapper.setProps({ codeFeedback: newCodeFeedback })
+
+    const rightCodes = wrapper.findAll('.right-code');
+    expect(rightCodes.length).toBe(1);
+    expect(rightCodes[0].element.id).toBe('word-2');
+
+    const wrongCodes = wrapper.findAll('.wrong-code');
+    expect(wrongCodes.length).toBe(1);
+    expect(wrongCodes[0].element.id).toBe('word-12');
+  });
+  test('Submit solution', async () => {
+    const submitButton = wrapper.find('.btn-success');
+    expect(submitButton.text()).toBe("Submit");
+    expect(submitButton.exists()).toBe(true);
+    
+    submitButton.trigger('click');
+    expect(wrapper.emitted().submitSolution[0]).toEqual([new Solution(1, [])]);
+
+    await wrapper.vm.$nextTick();
+
+    expect(wrapper.find('.btn-success').exists()).toBe(false);
+  })
+  test('Select and remove code word', async () => {
+    const selectedWord = 2;
+    const targetWordButton = wrapper.find('#word-' + selectedWord);
+    expect(targetWordButton.exists()).toBe(true);
+
+    targetWordButton.trigger('click');
+    await wrapper.vm.$nextTick();
+
+    const modal = wrapper.getComponent(BModal);
+    const buttons = modal.findAllComponents(BButton);
+    const submitButton = buttons.find((button) => button.text() == "Ok");
```

Comment:
I recommend a less error prone such as a specific style class, id, or attribute. But do not compare by text.
That is the equivalent to testing for a Go error that does not define an interface.

Reply:
```suggestion
    const submitButton = buttons.find((button) => button.attributes().class.includes("btn-primary"));
```
I can search by btn-primary class which is the Ok button. I do not see more possibilities for this.

Reply:
Far better than what you have now.